specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,175 @@
+@1 = internal constant [3 x i8] c"r1\00"
+@2 = internal constant [3 x i8] c"r2\00"
+
+; entry point definition
+
+define i64 @Entry_Point_Name() #0 {
+entry:
+
+  ; calls to QIS functions  
+  tail call void @__quantum__qis__h__body(%Qubit* null)
+  tail call void @__quantum__qis__cnot__body(%Qubit* null, %Qubit* inttoptr (i64 1 to %Qubit*))
+  tail call void @__quantum__qis__mz__body(%Qubit* null, %Result* null)
+  tail call void @__quantum__qis__mz__body(%Qubit* inttoptr (i64 1 to %Qubit*), %Result* inttoptr (i64 1 to %Result*))
+
+  ; calls to record the program output
+  tail call void @__quantum__rt__initialize_record_output(i8* getelementptr inbounds ([16 x i8], [16 x i8]* @0, i32 0, i32 0))  
+  tail call void @__quantum__rt__tuple_record_output(i64 2, i8* null)
+  tail call void @__quantum__rt__result_record_output(%Result* null, i8* getelementptr inbounds ([3 x i8], [3 x i8]* @1, i32 0, i32 0))
+  tail call void @__quantum__rt__result_record_output(%Result* inttoptr (i64 1 to %Result*), i8* getelementptr inbounds ([3 x i8], [3 x i8]* @2, i32 0, i32 0))
+
+  ret i64 0
+}
+
+; declarations of QIS functions
+
+declare void @__quantum__qis__h__body(%Qubit*)
+
+declare void @__quantum__qis__cnot__body(%Qubit*, %Qubit*)
+
+declare void @__quantum__qis__mz__body(%Qubit*, %Result*)
+
+; declarations of functions used for output recording
+
+declare void @__quantum__rt__initialize_record_output(i8*)
+
+declare void @__quantum__rt__tuple_record_output(i64, i8*)
+
+declare void @__quantum__rt__result_record_output(%Result*, i8*)
+
+; attributes
+
+attributes #0 = { "entry_point" "required_qubits"="2" "required_results"="2" }
+```
+
+TODO: do we need additional restrictions for output recording? Like e.g. it needs to be a dedicated block, or output recording functions can only occur in the last block of the entry point?
+
+TODO: do we need a __quantum__rt__initialize and __quantum__rt__finalize?
+TODO: do measurements need to be at the end of the quantum computations, or should the backend take care of deferring them if necessary?
+
+TODO: a profile identifier and version number within the IR would be good to have (may be needed for correct usage of qubit and result pointers)   
+-> look into custom target triples
+
+## Entry Point Definition
+
+The bitcode contains the definition of the LLVM function that should be invoked when the program is executed, 
+refered to as entry point in the rest of this profile specification. 
+The name of this function may be chosen freely, as long as it is a valid 
+[global identifier](https://llvm.org/docs/LangRef.html#identifiers) by LLVM standard. 
+
+The entry point may not take any parameters and must return an exit code in the form of a 64-bit integer. The exit code 0 must be used to indicate a successful execution of the program. 
+
+### Attributes
+
+The following custom attributes must be attached to the entry point function:
+
+- An attribute named "EntryPoint" that does not 
```

Comment:
What does it not?

Reply:
Name `EntryPoint` is inconsistent with demo program and later program, but consistent with the current convention.

Reply:
The doc is not updated and read for detailed review.